metadata/src/test/java/org/apache/kafka/controller/QuorumControllerTest.java

```diff
@@ -173,6 +176,121 @@ private void testDelayedConfigurationOperations(LocalLogManagerTestEnv logEnv,
+                assertEquals("Unable to replicate the partition " + replicationFactor + " time(s): All brokers " +
+                    "are currently fenced.", createTopicsResponseData.topics().find("foo").errorMessage());
+
+                // Unfence all brokers
+                sendBrokerheartbeat(active, brokerCount, brokerEpochs);
+                createTopicsResponseData = active.createTopics(
+                    createTopicsRequestData).get();
+                assertEquals(Errors.NONE.code(), createTopicsResponseData.topics().find("foo").errorCode());
+                Uuid topicIdFoo = createTopicsResponseData.topics().find("foo").topicId();
+                sendBrokerheartbeat(active, brokerCount, brokerEpochs);
+                Thread.sleep(sleepMillis);
+
+                // All brokers should still be unfenced
+                for (int brokerId = 0; brokerId < brokerCount; brokerId++) {
+                    assertTrue(active.replicationControl().isBrokerUnfenced(brokerId),
+                        "Broker " + brokerId + " should have been unfenced");
+                }
+                createTopicsRequestData = new CreateTopicsRequestData().setTopics(
+                        new CreatableTopicCollection(Collections.singleton(
+                            new CreatableTopic().setName("bar").setNumPartitions(1).
+                                setConfigs(new CreateableTopicConfigCollection(Collections.
+                                    singleton(new CreateableTopicConfig().setName("min.insync.replicas").
+                                        setValue("2")).iterator())).
+                                setReplicationFactor(replicationFactor)).iterator()));
+                createTopicsResponseData = active.createTopics(createTopicsRequestData).get();
+                assertEquals(Errors.NONE.code(), createTopicsResponseData.topics().find("bar").errorCode());
+                Uuid topicIdBar = createTopicsResponseData.topics().find("bar").topicId();
+
+                // Fence some of the brokers
+                TestUtils.waitForCondition(() -> {
+                        sendBrokerheartbeat(active, brokersToKeepUnfenced, brokerEpochs);
+                        for (int i = brokersToKeepUnfenced; i < brokerCount; i++) {
+                            if (active.replicationControl().isBrokerUnfenced(i)) {
+                                return false;
+                            }
+                        }
+                        return true;
+                    }, sessionTimeoutSec * 2 * 1000,
+                    "Fencing of brokers did not process within expected time"
+                );
+
+                // At this point the brokers we want fenced are fenced.
+                // Send another heartbeat to the brokers we want to keep alive
+                sendBrokerheartbeat(active, brokersToKeepUnfenced, brokerEpochs);
+                int[] expectedIsr = new int[brokersToKeepUnfenced];
```

Comment:
If we create the topic with an explicit assignment, then I think we can remove this logic to build the expected ISR since we would be able to assert the

Reply:
Given we are keeping only one broker to remain unfenced, the expected isr and leader are simply that broker. I kept this logic to allow for paramterization of the test. Given we are not using the parameters right now, will remove this logic.